pkg/kindsys/kind.cue

```diff
@@ -0,0 +1,106 @@
+package kindsys
+
+import (
+	"strings"
+
+	"github.com/grafana/thema"
+)
+
+// A Kind specifies a type of Grafana resource.
+//
+// An instance of a Kind is called an entity. An entity is a sequence of bytes -
+// for example, a JSON file or HTTP request body - that conforms to the
+// constraints defined in a Kind, and enforced by Grafana's entity system.
+//
+// Once Grafana has determined a given byte sequence to be an
+// instance of a known Kind, kind-specific behaviors can be applied,
+// requests can be routed, events can be triggered, etc.
+//
+// Classes and objects in most programming languages are analogous:
+//   - #Kind is like a `class` keyword
+//   - Each declaration of #Kind is like a class declaration
+//   - Byte sequences are like arguments to the class constructor
+//   - Entities are like objects - what's returned from the constructor
+//
+// There are four essential categories of kinds: Raw, Slot, CoreStructured and CustomStructured.
```

Comment:
slot vs custom structured 🤔 

external plugins will be both?

Reply:
Any plugin could offer both, though we might not want core plugins to do it.

The wordsmithing for `#Kind` is...really hard 😭